### PR TITLE
IamAwsProvider: fix custom endpoint having non-default port

### PR DESCRIPTION
**Pull request by jgolda**
Follow-up to https://github.com/minio/minio-java/issues/1522

### CODE REVIEW

**Comment by jgolda** [comment]
As to the pipeline errors they are most likely unrelated to my changes - same error occurs on master branch and the class i touched is not used during the testes, so it should not affect them. I manually tested the built library in an example app based on the FileUploader from project's README, just to authenticate and call an API method. 

As far as i recall the tests fail because min.io instance expected to run TLS is not running TLS, hence giving this error. 
I'll try to research it in the evening - if you have any ideas as to how to fix it, i'll also appreciate it

File: api/src/main/java/io/minio/credentials/IamAwsProvider.java [hunk 1]
```diff
@@ -140,6 +140,7 @@ private String fetchImdsToken() {
           new HttpUrl.Builder()
               .scheme(url.scheme())
               .host(url.host())
+              .port(url.port())
```
Comment (balamurugana) [inline api/src/main/java/io/minio/credentials/IamAwsProvider.java:143]: Add port only if port and service do not match i.e. ignore port when port and service matches i.e HTTP -> 80, HTTPS -> 443

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/minio/credentials/IamAwsProvider.java (hunk 1 above)
Comment (jgolda) [reply api/src/main/java/io/minio/credentials/IamAwsProvider.java:143]: Is there any argument to do it this way? To my best knowledge applying the default port doesn't affect the call in any negative way. Any additional logic makes this feature more complicated to understand and maintain for future developers. 

Ofc if you insist, i'll change it, but i'd love to see a rationale behind this

Comment (balamurugana) [reply api/src/main/java/io/minio/credentials/IamAwsProvider.java:143]: It is required for Signature v4 computation.

Comment (jgolda) [reply api/src/main/java/io/minio/credentials/IamAwsProvider.java:143]: Are IAM requests signed? I cannot recall seeing any signing in IamAwsProvider and cannot recall this in AWS docs. To my best knowledge these are plain HTTP requests

Comment (balamurugana) [reply api/src/main/java/io/minio/credentials/IamAwsProvider.java:143]: For `IamAwsProvider`, the request is not signed. However it is better to add port only if it is not standard port for correctness.

Comment (jgolda) [reply api/src/main/java/io/minio/credentials/IamAwsProvider.java]: As you wish. Uploaded new commit with the fix

File: api/src/main/java/io/minio/credentials/IamAwsProvider.java [hunk 2]
```diff
@@ -184,12 +201,9 @@ private HttpUrl getIamRoleNamedUrl(String token) {
...
     } else {
-      url =
-          new HttpUrl.Builder()
-              .scheme(url.scheme())
-              .host(url.host())
-              .addPathSegments("latest/meta-data/iam/security-credentials/")
-              .build();
+      HttpUrl.Builder urlBuilder = new HttpUrl.Builder().scheme(url.scheme()).host(url.host());
+      setPortIfRequired(urlBuilder, url);
+      url = urlBuilder.addPathSegments("latest/meta-data/iam/security-credentials/").build();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/credentials/IamAwsProvider.java:206]:
```suggestion
      url =
          new HttpUrl.Builder()
              .scheme(url.scheme())
              .host(url.host())
              .port(url.port())
              .addPathSegments("latest/meta-data/iam/security-credentials/")
              .build();
```

File: api/src/main/java/io/minio/credentials/IamAwsProvider.java [hunk 3]
```diff
@@ -136,12 +141,9 @@ private String fetchImdsToken() {
...
     } else {
-      url =
-          new HttpUrl.Builder()
-              .scheme(url.scheme())
-              .host(url.host())
-              .addPathSegments("latest/api/token")
-              .build();
+      HttpUrl.Builder urlBuilder = new HttpUrl.Builder().scheme(url.scheme()).host(url.host());
+      setPortIfRequired(urlBuilder, url);
+      url = urlBuilder.addPathSegments("latest/api/token").build();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/credentials/IamAwsProvider.java:146]:
```suggestion
      url =
          new HttpUrl.Builder()
              .scheme(url.scheme())
              .host(url.host())
              .port(url.port())
              .addPathSegments("latest/api/token")
              .build();
```

**Review by balamurugana — CHANGES_REQUESTED** [review]
As `okhttp3.HttpUrl` is taking care of default port for `http` and `https`, this commit should be reverted

**Review by balamurugana — APPROVED** [review]

**Review by minio-trusted — APPROVED** [review]

**Pull request merged**